### PR TITLE
Use common environment for releasing

### PR DESCRIPTION
**Pull request by weshaggard**
- Switch to using common package-publish environment for release
- Switch to using checkout instead of manual clone for azure-sdk-build-tools

### CODE REVIEW

**Review by benbp — APPROVED** [review]

File: eng/pipelines/templates/stages/archetype-sdk-client.yml [hunk 1]
```diff
@@ -125,5 +125,3 @@ stages:
         ArtifactName: packages
         TargetDocRepoOwner: ${{ parameters.TargetDocRepoOwner }}
         TargetDocRepoName: ${{ parameters.TargetDocRepoName }}
-        ${{ if eq(parameters.ReleaseDependsOnLiveTests, 'false') }}:
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-client.yml:128]: @pallavit @jsquire I just wanted to let you know I'm removing this feature that @hallipr added about a year ago which forced folks that disabled live tests for release to ignore the failures. This feature hasn't been used even once in the last year and given our loosing up on release approvers I'm removing this also. Please let me know if you have any concerns.

**Comment by weshaggard** [comment]
Fixes https://github.com/Azure/azure-sdk-tools/issues/5923

**Comment by weshaggard** [comment]
Test release pipeline https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3490066&view=logs&j=3517b659-d3e8-545e-d9c6-aff172e087bb

File: eng/pipelines/templates/stages/archetype-sdk-client.yml (hunk 1 above)
Comment (jsquire) [reply eng/pipelines/templates/stages/archetype-sdk-client.yml:128]: No objections from me.

**Pull request merged**